packages/ember-routing-views/lib/views/link.js

```diff
@@ -408,19 +409,11 @@ var LinkComponent = EmberComponent.extend({
       queryParams = {};
     }
 
-    if (attrs.disabledClass) {
-      this.set('disabledClass', attrs.disabledClass);
-    }
-
-    if (attrs.activeClass) {
-      this.set('activeClass', attrs.activeClass);
-    }
-
     if (attrs.disabledWhen) {
-      this.set('disabled', attrs.disabledWhen);
+      this.set('disabled', getValue(attrs.disabledWhen));
     }
 
-    var currentWhen = attrs['current-when'];
+    var currentWhen = getValue(attrs['current-when']);
 
     if (attrs.currentWhen) {
```

Comment:
416 and 418 feel somewhat misaligned. we should use currentWhen in both cases, or refactor to improve clarity of intent.